### PR TITLE
request-limit-increase action for account resource

### PR DESCRIPTION
**Pull request by scotwk**
Issue #971 

Questions:

- Should this work for all resources ('EC2', 'ELB', 'VPC', 'AutoScaling', 'RDS', 'EBS', 'SES', 'IAM')?  Only works for EBS right now
- Should the severity level be configurable and if so what should the default be?  If not, what should it always be set to?  Choices are `urgent`, `high`, `normal`, and `low`
- Should the code be `volume-issue`?  (Other choices are `other`, `apis`, `feature-request`, `general-guidance`)

### CODE REVIEW

**Comment by kapilt** [comment]
ideally all resources, although could drop ses and iam i suspect wrt to auto limit increases.

re code, i think general-guidance, but i'll have to double check.

severity, config seems good, default to normal

File: c7n/resources/account.py [hunk 1]
```diff
@@ -434,3 +434,59 @@ def process(self, resources, event=None):
...
+    default_template = 'Please raise the limit of {id} by {percent}%'
+    
+    def process(self, resources):
+        session = local_session(self.manager.session_factory)
+        client = session.client('account')
+        
+        for resource in resources:
+            resource_id = resource.get('VolumeId'),
+
+            subject = 'Raise the limit of {}'.format(resource_id)
```
Comment (kapilt) [inline c7n/resources/account.py]: this seems a bit odd, not sure why volumeid would be present, its an account limit we want to increase, not that of the individual volume which is self service. ie let's say i default to 50tb data limit on the account, and i do that spread across a 100 500gb volumes, i want to increase the total volume of the account, not that of an individual volume. the resource here is the actual account itself.

Comment (scotwk) [inline c7n/resources/account.py]: that was just a placeholder while I was waiting on answers from you.  Thank you for the explanation.

**Comment by scotwk** [comment]
@kapilt - When I use the service-limit filter I don't see ELB as a service in any of the checks.  I do see CloudFormation.  Are those the same thing?

**Comment by scotwk** [comment]
@kapilt - good to go, pending the three questions I put into the code review

**Comment by scotwk** [comment]
remove emails from test placebo json files

**Comment by scotwk** [comment]
@kapilt - scrubbed.  Awaiting your response to code review.  Thanks

File: c7n/resources/account.py [hunk 2]
```diff
@@ -434,3 +434,83 @@ def process(self, resources, event=None):
...
+        'request-limit-increase',
+        **{'notify': {'type': 'array'},
+           'percent-increase': {'type': 'number'},
+           'subject': {'type': 'string'},
+           'message': {'type': 'string'},
+           'severity': {'type': 'string', 'enum': ['urgent', 'high', 'normal', 'low']},
+           'required': ['percent-increase'],
+           })
+
+    permissions = ('support:DescribeTrustedAdvisorCheckResult',)
```
Comment (kapilt) [inline c7n/resources/account.py]: support:CreateCase  from iam policy generator

File: c7n/resources/account.py [hunk 3]
```diff
@@ -434,3 +434,83 @@ def process(self, resources, event=None):
...
+
+    default_subject = 'Raise the account limit of {service}'
+    default_template = 'Please raise the account limit of {service} by {percent}%'
+    default_severity = 'normal'
+
+    service_code_mapping = {
+        'AutoScaling': 'auto-scaling',
+        'CloudFormation': 'elastic-load-balancing',  # Is this the correct service name?
```
Comment (scotwk) [inline c7n/resources/account.py]: Fixed

File: c7n/resources/account.py [hunk 4]
```diff
@@ -434,3 +434,83 @@ def process(self, resources, event=None):
...
+        'EBS': 'amazon-elastic-block-store',
+        'EC2': 'amazon-elastic-compute-cloud-linux',
+        'RDS': 'amazon-relational-database-service-aurora',
```
Comment (scotwk) [inline c7n/resources/account.py]: Per Kapil - this is the best we can do

**Comment by scotwk** [comment]
@kapilt - good to go

**Comment by kapilt** [comment]
merge conflict needs resolving before merge.

**Comment by scotwk** [comment]
@kapilt - resolved

**Pull request merged**